reconnecting-eventsource.d.ts

```diff
@@ -0,0 +1,11 @@
+export interface ReconnectableEventSourceInit extends EventSourceInit {
+	// the maximum time to wait before attempting to reconnect in ms, default `3000`
+	// note: wait time is randomised to prevent all clients from attempting to reconnect simultaneously
+	max_retry_time?: number;
+}
+
+export default class ReconnectingEventSource extends EventSource {
+	constructor(url: string, configuration?: ReconnectableEventSourceInit);
```

Comment:
I would like to suggest this to match the signature of `EventSource`

```suggestion
	constructor(url: string | URL, configuration?: ReconnectableEventSourceInit);
```

Reply:
Your code does not support type URL as first argument, so I think no. For example .indexOf is missing method of URL

Reply:
You must refactor this https://github.com/fanout/reconnecting-eventsource/blob/master/src/reconnecting-eventsource.js#L54

Reply:
Ah I see.  Okay, good point.

[MDN also says this is just a string](https://developer.mozilla.org/en-US/docs/Web/API/EventSource/EventSource) so I say let's accept this for now and we can do that as a separate issue.